### DISCUSSION TITLE
UCB CS188: Introduction to Artificial Intelligence - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#132] [Announcements]
# UCB CS188: Introduction to Artificial Intelligence - CS自学指南

CS自学指南

https://csdiy.wiki/%E4%BA%BA%E5%B7%A5%E6%99%BA%E8%83%BD/CS188/

**Comment by SeeYouCowboi** [comment]
请问为什么我注册gradescope的时候它说这个课程的ID—93PWD8无效

**Reply by SeeYouCowboi** [reply]
成功了，学校改成UC berkly就好了

**Reply by syccxdr** [reply]
我也遇到了这个问题 把学校改成了 2U uc伯克利也不行

**Reply by YangHuaLuoJinZiGuiTi** [reply]
berkeley有很多选择， 要UC berkely才行

**Reply by Tina-LT** [reply]
这位同学你看的18还是22的课呀

**Reply by jaco1b** [reply]
谁能教教我怎么注册么

**Comment by hangfrieddays** [comment]
选学校选的是2U UC Berkeley还是无效，是因为93PWD8这个课程吗无效吗

**Reply by hangfrieddays** [reply]
成功了，注意2U UC berkely 和UC berkely不一样

**Reply by Abyssal-si** [reply]
请问课程码是在哪里看的？

**Comment by greatxue** [comment]
请问这个和机器学习选哪个上？

**Reply by Mamihla** [reply]
时间多就先上这个吧，覆盖的内容比较多，最后那几节课也讲了一点机器学习的内容

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by hangfrieddays** [comment]
这两个开课目标不一样吧，cs188类似一个大纲课，介绍了很多人工智能相关的知识，包括机器学习，如果有时间建议从这个课学起，没有足够时间其实从吴恩达的机器学习开始看也没有问题。



---原始邮件---
发件人: "Kevin ***@***.***&gt;
发送时间: 2023年6月20日(周二) 下午2:12
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [PKUFlyingPig/cs-self-learning] UCB CS188: Introduction to Artificial Intelligence - CS自学指南 (Discussion #132)




 
请问这个和机器学习选哪个上？

**Reply by Sunstreamy** [reply]
你好，想问一下，没有先学cs70，但是对离散数学有一小部分基础，推荐直接学这个吗，还是说先学cs50ai好呢？

**Comment by chenchenchenxyz** [comment]
这个notes在resources里吗

**Reply by YangHuaLuoJinZiGuiTi** [reply]
你看看它的Calendar，里面啥都有

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by doubl3surprise** [comment]
请问我如何从Gradescope提交我的project，我只找到了交homework的地方

**Reply by oneHFR** [reply]
我也是找不到project的提交地址，生成的token往哪里上传呀？

**Reply by rinevard** [reply]
project似乎只有本地测试，运行
python autograder.py
即可

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by rinevard** [comment]
对在学CS188 fa18的同学的一些小提醒：

1. CS188 fa18关于贝叶斯网络的notes没有其它那些notes质量高，可以考虑看看CS188 fa23的关于贝叶斯网络的notes（更加详细）和COMP538: Introduction to Bayesian Networks（更加深入）作为补充。

2. CS188 fa18的project4中并没有让我们实现贝叶斯网络中的推断，可以先做CS188 fa23的project4的前几个与贝叶斯网络相关的题，再回来做fa18的project4.

**Reply by lxj-bzm** [reply]
同学，fa23有公开的课程注册码吗？谢谢！

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by hangfrieddays** [comment]
抱歉，我没有嘞，我记得之前在一亩三分地上找到过当时的注册码，不过现在忘记了



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年4月3日(周三) 晚上6:35
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [PKUFlyingPig/cs-self-learning] UCB CS188: Introduction to Artificial Intelligence - CS自学指南 (Discussion #132)




 
同学，fa23有公开的课程注册码吗？谢谢！

**Comment by 1nram1** [comment]
请问 cs50 Al
和CS188 建议上哪一个？

**Reply by hqhq1025** [reply]
我觉得cs50ai对于新手更加友好，如果觉得cs188有点难度可以先学cs50ai。上了cs188感觉没必要学cs50ai了

**Reply by Sunstreamy** [reply]
你好，想问一下，没有先学cs70，但是对离散数学有一小部分基础，推荐直接学这个吗，还是说先学cs50ai好呢？

**Reply by hqhq1025** [reply]
我觉得 cs70 跟 cs188 的关联似乎不是很强吧,  我觉得无所谓. 你可以先听听 188 看看对你的难易, 略难的话再去看 cs50ai

**Reply by puji4810** [reply]
直接188完事了

**Reply by puji4810** [reply]
> 直接188完事了

不会的问ai

**Reply by Sunstreamy** [reply]
好的，谢谢各位的建议，另外，对于课程的版本，选取2018的还是最新版本呢（2025sp？）我看网上似乎认为2018版要更加出色（貌似是授课教师十分著名）

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by liangchen68** [comment]
官网课程怎么打不开了

**Reply by Wangch29** [reply]
有时候会更新，一般要一两天

**Comment by sarendis56** [comment]
个人听了一圈还是喜欢fall23的版本，两个老师都讲的很好，也是最新的公开版本：https://inst.eecs.berkeley.edu/~cs188/fa23/. Peyrin基本每个学期都讲，和sp23的版本有细微差距（改进），主要是sp前面informed search部分讲太多，然后之后Russell教授讲课又太慢导致后面一系列内容个人都感觉有点赶

**Reply by alexACxx** [reply]
请问下现在哪里还能看呀？

**Comment by study-overflow** [comment]
为什么现在点进课程链接之后是CalNet Authentication Service的界面，需要登录啊……是不是课程被锁了？求助大佬

**Reply by f3f3x0** [reply]
遇到同样问题

**Reply by zyy98** [reply]
遇到了一样的问题，不知道这种情况是一两天内自我修复，还是永久lock了QuQ
课程bilibili有人上传，但是不知道有没有同学可以分享一下slides+project

**Reply by ultrabababa** [reply]
同问

**Reply by sarendis56** [reply]
已经分享在评论区下面了

**Comment by sarendis56** [comment]
我在写上一条的时候还是公开的...我上传了一下我用的资源，包括slides，notes，discussion，project starters等等，在https://github.com/Hyle33ies/CS188。需要的同学可以下载，如果课程老师联系我了我会删掉

**Reply by study-overflow** [reply]
好人一生平安！

**Reply by zyy98** [reply]
太感谢了！！！

**Reply by zyy98** [reply]
不抱希望的问一下有么有小伙伴有22fall的课件，个人看的是这个版本，因为youtube上有章节分段，对我来说条理清晰一些，似乎两年的slides差别不大但是还是略有不同...当然了已经有朋友给了23的资源非常感谢了🙏

**Reply by sarendis56** [reply]
https://web.archive.org/web/20230106141521/https://inst.eecs.berkeley.edu/~cs188/fa22/
可能有点慢，但是看了一下能用

**Reply by zyy98** [reply]
真的太感谢了！！！

**Reply by Zephyr-Yara** [reply]
现在在哪能看到CS 188的recording呀

**Reply by AnaBotTech** [reply]
你上面那个人给的链接就能看：https://web.archive.org/web/20230106141521/https://inst.eecs.berkeley.edu/~cs188/fa22/，以及b站上有可能有我没去看过

**Reply by Wallaby-Wang** [reply]
有朋友可以分享一下23的课件资源notes,slides吗，一直看人说但没找到🙏

**Comment by jrryzh** [comment]
2024.6.3好像课程网站打不开了 需要登陆ucb学校账号

**Reply by LetMyTearsFlow** [reply]
有没有能拿到这个账号的方法啊。。。

**Reply by Wangch29** [reply]
当然有方法。你去加州大学伯克利分校官网，点击admission，然后提交一些你的个人资料，再支付260美元。过几段时间，他们有就可能发给一封邮件，同时会给你账户。

**Reply by Mamihla** [reply]
要是没有给账户可以退款吗 ...

**Reply by Wangch29** [reply]
这个是申请ucb入学的流程。。。不能退款

**Reply by Sakura-Flowing** [reply]
话说问问ucb的大部分课是都要永久关闭权限嘛，还是只是近段时间关闭权限，（话说什么原因呀，有佬知道嘛）

**Comment by Sakura-Flowing** [comment]
话说问问ucb的大部分课是都要永久关闭权限嘛，还是只是近段时间关闭权限，（话说什么原因呀，有佬知道嘛）

**Reply by rinevard** [reply]
ucb的课似乎是今年五月左右开始大规模关闭。我当时查了下原因，似乎是美国的一个残障方面的组织认为伯克利的公开课的无障碍化做得不够好，然后和伯克利打官司，伯克利输了，于是就有了[这份Consent Decree](https://www.justice.gov/d9/case-documents/attachments/2022/11/21/consent_decree_-_u.s._v._uc_berkeley.pdf). 主要时间节点如下：

生效日期: 2022年12月2日

4个月内(2023年4月2日前): 制定网页无障碍程序

9个月内(2023年9月2日前): BerkeleyX平台和新音频视频内容合规

18个月内(2024年6月2日前): 网站和子域名合规

36个月内(2025年12月2日前): 现有音频和视频内容合规 

伯克利选择让课程不再开放，好像这样就不用做无障碍化了。

**Comment by Aniurm** [comment]
2024.6.18
喜大普奔，CS188 sp24开放啦

**Comment by AAAAyl0n** [comment]
几个往年的课程网站现在应该进不去了，但是2024是开放的：https://inst.eecs.berkeley.edu/~cs188/sp24/

**Reply by VictorDonkey** [reply]
24年的有课程录屏吗？

**Reply by puji4810** [reply]
24的也看不了了，悲

**Reply by puji4810** [reply]
不对，可以看，抱歉

**Reply by LingeringAutumn** [reply]
请问课程码是什么？我一点进去它就要我输课程码

**Comment by Linmj-Judy** [comment]
有AI和ML基础，做过ML应用的科研，请问这个和CS229推荐先上哪个呀？

**Reply by sarendis56** [reply]
没有先后顺序 但是如果对229内容不熟悉的话优先上229 未来用到的可能性大一些

**Reply by rinevard** [reply]
CS188学的东西很基础，个人认为偏向传统方法。如果有AI和ML基础就不必学这个了，好奇的话看看notes就行

**Reply by oplisty** [reply]
在哪里看note啊，，课程网站好像锁了

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by zhou427** [comment]
问下各位教材有必要仔细看吗？还是只看notes也足够完成作业和project？

**Reply by EntroShape** [reply]
视频可以3倍速，不费多少时间

**Comment by Yh1zzzzzzz** [comment]
建议看18fall的教学视频，最新的23fall其中一位老师讲的比较烂

**Comment by darstib** [comment]
请问有比较新的 project 的 solution 吗？ 我爬完了楼好像是没看见，在github上找到了最新的是
https://github.com/szzxljr/CS188_Course_Projects
不知道是不是一样的？

**Reply by sarendis56** [reply]
cs188据我观察每年project没什么差别 做哪年貌似差不多吧

**Reply by darstib** [reply]
噢噢，好的谢谢，主要是有一些project有改动

![image](https://github.com/user-attachments/assets/6bd53ba0-3bfb-4c72-a635-5e96fd478129)

不过他们考虑的还真周到

**Reply by Boylees** [reply]
请问你们是怎么看课的呢，我这边需要伯克利的账号登录才能看

**Reply by darstib** [reply]
sp24 当前开放了 https://inst.eecs.berkeley.edu/~cs188/sp24/
如果依旧关闭可以来看看我的 https://github.com/Darstib/cs188/
我正在梳理学习，做project，需要的文件都先下下来了，怕之后需要认证

**Reply by darstib** [reply]
如果你是说视频课，那我就不知道了

**Reply by XiaoPeng0x3** [reply]
您好，这门课可以提交OJ吗？还是说本地测试就足够了？

**Reply by darstib** [reply]
我不清楚，我只是在本地测试

**Reply by AnaBotTech** [reply]
十分感谢，已经保存并star了

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by AcuLY** [comment]
24春季的已经关闭了，现在秋季的是开放的 https://inst.eecs.berkeley.edu/~cs188/fa24/

**Reply by Rcpentuy** [reply]
sp24好像没有关闭吧，我这里还能打开

**Comment by Zephyr-Yara** [comment]
CS 188 对Python的要求高嘛，之前学过C，直接学CS 188会不会有困难

**Comment by Zephyr-Yara** [comment]
有学CS 188 24 Spring的嘛，project0 python安装就出现了问题

**Reply by wang-jiahao** [reply]
我记得需要安装3.11之前的版本，太新的不行

**Comment by wang-jiahao** [comment]
目前学到 project4,挂一个我的仓库，欢迎交流讨论（qq：2328796228，email：jiahaowang@smail.nju.edu.cn）
https://github.com/wang-jiahao/CS188

**Comment by JourneyForth** [comment]
想请问一下，点击Fall 2022的链接跳转到课程网站需要登录伯克利帐户这个怎么办啊？

**Reply by bretonyang** [reply]
可以用wayback machine試試看

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by Wizcoo** [comment]
2025课程地址：https://inst.eecs.berkeley.edu/~cs188/sp25/

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by Zlatanwic** [comment]
现在好像是需要伯克利的账户怎么办啊兄弟们

**Comment by darstib** [comment]
自荐一下自己基本完成的自学（sp24版本，笔记摘要+project简要思考过程+代码）

- 笔记：http://darstib.github.io/blog/note/cs188
- 项目仓库：https://github.com/darstib/cs188

最后一个project 的最后一个检查点没能通过，调参数没有太看明白，欢迎 [联系交流](https://darstib.github.io/connect_me/)

**Reply by jaco1b** [reply]
想问一下啊这个学校账号怎么搞啊

**Reply by darstib** [reply]
可以看 https://github.com/PKUFlyingPig/cs-self-learning/discussions/132#discussioncomment-10482389 的评论，这一点我没有去解决，因为我上的时候网站都是可行的，并没有问题。在上面也有对账户的讨论（应该是需要交学费）

**Comment by fjc-hub** [comment]
这是我写过的Course中工程质量最差👎的。比CS61A都差了十万八千里，纯纯大粪

**Reply by fjc-hub** [reply]
好吧，也还行😁

**Reply by j4deng** [reply]
哈哈哈哈哈

**Comment by wzciczw** [comment]
最近有没有一起学CS188的小伙伴，本人已完成Project1，希望可以找到一起学习这门课的朋友交流~qq 1157384292

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by guojiayu-ai** [comment]
2025版本https://inst.eecs.berkeley.edu/~cs188/sp25/

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by Harry-Just-ZJU** [comment]
请问各位佬calendar上的recording和videos有啥区别啊 看哪个合适啊

**Reply by Sunstreamy** [reply]
recording是往年的授课内容，recording是该学期的回放

**Comment by YuTaki23** [comment]
请问各位 sp24版本有开放gradescope么 如果有的话 注册码是多少

**Reply by Sunstreamy** [reply]
请问一下为什么不看最新版25sp呢？

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by iDako7** [comment]
具体的邀请码在 policy 里面会有的：https://inst.eecs.berkeley.edu/~cs188/sp25/policies/

**Comment by ctfy66** [comment]
课程网站无法打开, 意外终止了连接, 检查网络没有问题,正常访问外网.

**Reply by ctfy66** [reply]
现在已经能够正常访问.

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by EnzoDing-rgb** [comment]
没有数学基础能学吗？

**Reply by 1010fzp0101** [reply]
学不了的， 这需要离散， 概率的知识

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by hangfrieddays** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我已经收到你的邮件了，我会及时回复你的

**Comment by zhaoboxuan92-cmd** [comment]
感觉非常充实，感谢CS188
